delivery/media/cat/examples/token-verification/main.js

```diff
@@ -0,0 +1,196 @@
+              logger.log("exp_extension: %s", exp_extension)
+              logger.log("exp_deadline: %s", exp_deadline)
+              logger.log("exp: %s", exp)
+              // support for cookie and header renewable type
+              if ((renewalType === 1 || renewalType === 2)  && exp_extension !== undefined && exp !== undefined) {
+                let lowT;
+                if (exp_deadline !== undefined) {
+                  lowT = exp - exp_deadline * 60
+                } else {
+                  lowT = exp - 1 * 60 // 1 mins renewal window by default
+                }
+                const now = Math.floor(Date.now()/1000);
+                logger.log("lowT: %s", lowT)
+                logger.log("now: %s", now)
+                // renew the token
+                if (now >= lowT && now < exp) {
+                  const new_exp = now + exp_extension
+                  catJSON.payload.set(ClaimsLabelMap.exp, new_exp)
+                  catJSON.payload.set(ClaimsLabelMap.iss, 'akamai.com')
+                  catJSON.payload.set(ClaimsLabelMap.iat, now)
+                  // renewal token is signed with different key
+                  catJSON.header.u.set(HeaderLabelMap.kid, new TextEncoder().encode('akamai_key_es256'))
+                  // change algo
+                  catJSON.header.p.set(HeaderLabelMap.alg, AlgoLabelMap.ES256);
+                  const esSignKey = await crypto.subtle.importKey(
+                    'jwk',
+                    es256PrivJwk,
+                    {
+                      name: 'ECDSA',
+                      namedCurve: 'P-256'
+                    },
+                    false,
+                    ['sign']
+                  );
+                  const cwtTokenBuf = await CWTGenerator.sign(catJSON.payload, { key: esSignKey }, catJSON.header, { isCoseCborTagAdded: true, isCWTTagAdded: true });
+                  const cwtTokenBase64 = base64url.encode(new Uint8Array(cwtTokenBuf));
+                  request.setVariable('PMUSER_RENEWED_CAT', cwtTokenBase64);
+                  request.setVariable('PMUSER_RENEWAL_TYPE', renewalType);
+                }
+              }
+              // Proceed and return the content
+              request.route({ query: finalQs })
+            } else {
+              request.respondWith(401, {}, result.errMsg)
+            }
+          } catch(error) {
+            request.respondWith(401, {}, 'Common access token signature verification failed')
+          }
+        } else {
+          // Token is not well formed. (i.e syntax errors)
+          request.respondWith(401, {}, result.errMsg)
+        }
+      } catch(error) {
+        request.respondWith(401, {}, error.message)
+      }
+    } else {
+      request.respondWith(403, {}, 'Common access token is not found in cookie {name=\'Common-Access-Token\'} or query string {cat=<token>}')
+    }
+  }
+}
+
+// export function onClientResponse (request, response) {
```

Comment:
Should there be an explanation why this is commented out?

Reply:
Added explanation. Thanks!